### PR TITLE
Added recursive query parameter and fixed some linting errors

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

**Review by tasherif-msft — APPROVED** [review]

File: cmd/syncProcessor.go [hunk 1]
```diff
@@ -471,9 +475,11 @@ func (b *remoteResourceDeleter) delete(object StoredObject) error {
 					}
 				}
 			case common.ELocation.BlobFS():
-				directoryClient := common.CreateDatalakeDirectoryClient(objectURL.String(), b.credInfo, nil, b.clientOptions)
-				// TODO : Recursive delete
-				_, err = directoryClient.Delete(ctx, nil)
+				clientOptions := b.clientOptions
```
Comment (adreed-msft) [inline cmd/syncProcessor.go:478]: This is a workaround for a lack of support for a continuation token, right? Is there any ongoing work to fix that up?

**Review by adreed-msft — COMMENTED** [review]
lgtm

**Review by adreed-msft — APPROVED** [review]

File: cmd/syncProcessor.go (hunk 1 above)
Comment (gapra-msft) [reply cmd/syncProcessor.go:478]: this is a workaround for lack of recursive support. The continuation token I believe will be hidden behind the SDK

Comment (gapra-msft) [reply cmd/syncProcessor.go:478]: The plan is not to support recursive=false for directories by design in the SDK, so we need to do this to work around that

**Pull request merged**